tmt/steps/prepare/artifact/providers/__init__.py

```diff
@@ -195,6 +199,201 @@ def _filter_artifacts(self, exclude_patterns: list[Pattern[str]]) -> Iterator[Ar
                 yield artifact
 
 
+class Repository:
+    """A class to represent a dnf/yum software repository."""
+
+    def __init__(self, logger: tmt.log.Logger, name: str, content: str):
+        self.logger = logger
+        self.name = name
+        self.content = content
+        self._repo_ids = self._get_repo_ids()
+
+    # The @overload decorator is used here to provide type hinting for multiple possible
+    # call signatures of the 'create' class method. This helps static type checkers like
+    # mypy understand that the method can be called in different ways (e.g., with 'url',
+    # 'file_path', or 'content'), even though Python doesn't support true function overloading.
+    # This specific overload defines the signature when creating a Repository from a URL.
+    @overload
+    @classmethod
+    def create(
+        cls,
+        logger: tmt.log.Logger,
+        *,
+        name: Optional[str] = None,
+        url: str,
+        file_path: None = None,
+        content: None = None,
```

Comment:
Can you not remove `file_path` and `content`? 

Also need to verify that if mypy does not match any of it it would error out. It might require stripping the annotations of the implementation, not familiar with the nuances of that, but some light experimentation should uncover the behavior